src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -898,7 +899,11 @@ private void ResetTabDataLoadFlags()
         private void RefreshInstalledAndUpdatesTabs()
         {
             // clear existing caches
-            Model.CachedUpdates = null;
+            if (Model.CachedUpdates != null)
+            {
+                Model.CachedUpdates.Packages = null;
```

Comment:
I'm skeptical this is accomplishing anything. Did you test before/after this change? What impact did it have?

Reply:
Maybe this one not much help. I put it first this one by tracking memory retention path. But next line `Model.CachedUpdates = null;` really reduces memory footprint after GC run.